File: forum-api-back/forum-api-back/DataAccess/DataObjects/Topic.cs
using System;
using System.Collections.Generic;

namespace forum_api_back.DataAccess.DataObjects
{
    public partial class Topic
    {
        public Topic()
        {
            Comments = new HashSet<Comment>();
        }

        public int Idtopic { get; set; }
        public string Titre { get; set; } = null!;
        public string NomUtilisateur { get; set; } = null!;
        public DateTime DateCreation { get; set; }
        public DateTime? DateModification { get; set; }

        public virtual ICollection<Comment> Comments { get; set; }

    }
}

File: forum-front/src/app/models/topic.ts
import { Comment } from "./comment";

export class Topic {

    idTopic: Number;
    titre: string;
    nomUtilisateur: string;
    dateCreation: Date;
    dateModification?: Date;
    comments: Comment[];

    constructor(idTopic: Number, titre: string, nomUtilisateur: string, dateCreation: Date, dateModification: Date, comments: Comment[]){
        this.idTopic = idTopic;
        this.titre = titre;
        this.nomUtilisateur = nomUtilisateur;
        this.dateCreation = dateCreation;
        this.dateModification = dateModification;
        this.comments = comments;
    }

}